### PR TITLE
feat: partition all tables by year and month

### PR DESCRIPTION
**Pull request by Ian2012**
### Description

This PR partitions almost all tables by year and month.

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @Ian2012! Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


This is currently a draft pull request. When it is ready for our review and all tests are green, click "Ready for Review", or remove "WIP" from the title, as appropriate.

File: models/grading/grading_events.sql [hunk 1]
```diff
@@ -3,7 +3,8 @@
     schema=env_var('ASPECTS_XAPI_DATABASE', 'xapi'),
     engine=get_engine('ReplacingMergeTree()'),
     primary_key='(org, course_key, verb_id)',
-    order_by='(org, course_key, verb_id, emission_time, actor_id, object_id, scaled_score, event_id)'
+    order_by='(org, course_key, verb_id, emission_time, actor_id, object_id, scaled_score, event_id)'}
```
Comment (bmtcril) [inline models/grading/grading_events.sql:6]: Looks like an extra close brace here

**Review by bmtcril — CHANGES_REQUESTED** [review]
Looks like one little issue, otherwise 👍

**Review by bmtcril — APPROVED** [review]
👍

**Comment by Ian2012** [comment]
@bmtcril should I bump version?

**Comment by bmtcril** [comment]
This one isn't version managed in code, we just bump the tag / release after merging.

**Comment by openedx-webhooks** [comment]
@Ian2012 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/aspects-dbt&entry.867055334=https://github.com/openedx/aspects-dbt/pull/42&entry.1484655318=https://github.com/Ian2012&entry.752974735=2024-01-26+21:13&entry.1917517419=2024-01-30+19:39&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**